make/clean.mk

```
@@ -43,13 +49,14 @@ clean-e2e-resources: clean-users clean-toolchain-namespaces-in-e2e clean-cluster
 clean-toolchain-namespaces-in-dev:
 	$(Q)oc delete namespace ${DEV_HOST_NS} || true
 	$(Q)oc delete namespace ${DEV_MEMBER_NS} || true
+	$(Q)oc delete namespace ${DEV_SSO_NS} || true
```

to avoid problems with finalizers, would it be a good idea to add `clean-toolchain-dev-sso-resources` as a dependency here?

```suggestion
clean-toolchain-namespaces-in-dev: clean-toolchain-dev-sso-resources
	$(Q)oc delete namespace ${DEV_HOST_NS} || true
	$(Q)oc delete namespace ${DEV_MEMBER_NS} || true
	$(Q)oc delete namespace ${DEV_SSO_NS} || true
```

---

Good point, but TBH I'm not 100% sure. I added `clean-toolchain-dev-sso-resources` here https://github.com/codeready-toolchain/toolchain-e2e/pull/996/files#diff-4940ce933c6483bbf87f37d0041ad7587d86cac23b65179f97f1e7a70b6e6d9bR59 , before invoking `clean-toolchain-namespaces-in-dev`. That should do, since it's the only place where we use it.

---

Having both of them as `clean-dev-resources` dependency, will make it work fine, i.e. it won't hang on keycloak resources deletion; but if we invoke directly `clean-toolchain-namespaces-in-dev` it would likely hang. Introducing the `clean-toolchain-dev-sso-resources` dependency to `clean-toolchain-namespaces-in-dev` will avoid this behavior.

As it is possible for the user to invoke directly `clean-toolchain-namespaces-in-dev`, it makes sense to me to add it.

---

Ok I see, it could hang if the env was deployed with `DEV_SSO=true` indeed, but users should use `clean-dev-resources` and not `clean-toolchain-namespaces-in-dev` I think. Anyways moving it solves this problem , done in https://github.com/codeready-toolchain/toolchain-e2e/pull/996/commits/e797e4d357c535f55480be116b8ee8f42fda5a07 . Thanks!